source/agora/common/BlockStorage.d

```diff
@@ -113,73 +113,37 @@ public class BlockStorage
 
         If it was mapping the same file, just return.
         If it was mapping the other file, close previously mapped file.
-        If a size change occurs, open it again.
 
         Params:
             findex = the index of the file.
-            resize = The size of the file to change. If 0 does not change it.
 
     ***************************************************************************/
 
-    private void map (size_t findex, size_t resize = 0) @trusted
+    private void map (size_t findex) @trusted
     {
-        // It's already mapped,
-        if ((this.file !is null) && (findex == this.file_index))
+        try
         {
-            // if it doesn't change its size
-            if (resize == 0)
-                return;
-
-            // if can use reserved memory
-            if ((this.data_size < resize) &&
-                (HeadSize + resize < this.getFileLength()))
-            {
-                this.writeDataLength(resize);
-                this.data_size = resize;
+            if ((this.file !is null) && (findex == this.file_index))
                 return;
-            }
-        }
 
-        if (this.file !is null)
-            this.release();
+            if (this.file !is null)
+                this.release();
 
-        this.file_index = findex;
-        const string name = this.getFileName(this.file_index);
+            this.file_index = findex;
 
-        if (resize != 0)
-        {
-            this.file =
-                new MmFile(
-                    name,
-                    MmFile.Mode.readWrite,
-                    resize + HeadSize + ReserveSize,
-                    null
-                );
-            this.writeDataLength(resize);
-            this.data_size = resize;
-        }
-        else if (name.exists())
-        {
             this.file =
                 new MmFile(
-                    name,
+                    this.getFileName(this.file_index),
                     MmFile.Mode.readWrite,
-                    0,
+                    MapSize,
                     null
                 );
-            this.data_size = this.readDataLength();
+
+            this.file_base = MapSize * this.file_index;
         }
-        else
+        catch (Exception ex)
```

Comment:
So the failure does not get reported to the caller. Does that mean we are okay with this component failing ?

Reply:
fixed

Reply:
fixed